egui/src/widgets/plot/items/values.rs

```diff
@@ -140,22 +147,45 @@ impl Default for Orientation {
 
 // ----------------------------------------------------------------------------
 
-#[derive(Default)]
-pub struct Values {
-    pub(super) values: Vec<Value>,
-    generator: Option<ExplicitGenerator>,
+pub enum PlotPoints<'v> {
```

Comment:
Good improvement, but I think it would be nice to also add a docstring

```suggestion
/// Represents many [`PlotPoint`]s.
///
/// These can be a borrowed slice, an owned `Vec`, or generated with a function.
pub enum PlotPoints<'v> {
```